README.md

```diff
@@ -24,14 +21,77 @@ If you have downloaded the repo without using the `--recurse-submodules` argumen
 git submodule update --init --recursive
 ```
 
+## coreMQTT Agent Library Configurations
+
+The MQTT Agent library exposes build configuration macros that are required for building the library.
+However, the MQTT Agent library does not include its own configuration file, but exposes configurable macros and their default values at the top of [core_mqtt_agent.h](source/include/core_mqtt_agent.h) and [core_mqtt_agent_command_functions.h](source/include/core_mqtt_agent_command_functions.h). Also, the library inherits any macros defined in coreMQTT's `core_mqtt_config.h`.
+
+To provide custom values for the configuration values, they must be either:
+* Defined in `core_mqtt_config.h` used by coreMQTT
+**OR**
+* Passed as compile time preprocessor macros
+
+## Building the Library
```

Comment:
Think this can be simplified too.  "Build the files in the x and y directories, and add z directory to your compiler's include path".  Then mention there is a cmake file for those using cmake.

Reply:
Added a sentence describing the source and include paths